src/job.rs

```diff
@@ -288,25 +288,28 @@ impl Job {
                             // Yandex error "554 5.7.1 [2] Message rejected under suspicion of SPAM; https://ya.cc/..."
                             // should definitely go here, because user has to open the link to
                             // resume message sending.
-                            let msg_id = MsgId::new(self.foreign_id);
-                            message::set_msg_failed(context, msg_id, Some(err.to_string())).await;
-                            match Message::load_from_db(context, msg_id).await {
-                                Ok(message) => {
-                                    chat::add_info_msg(context, message.chat_id, err.to_string())
-                                        .await
-                                }
-                                Err(e) => error!(
-                                    context,
-                                    "couldn't load chat_id to inform user about SMTP error: {}", e
-                                ),
-                            };
                             Status::Finished(Err(format_err!("Permanent SMTP error: {}", err)))
                         }
                     }
                     async_smtp::smtp::error::Error::Transient(_) => {
                         // We got a transient 4xx response from SMTP server.
                         // Give some time until the server-side error maybe goes away.
-                        Status::RetryLater
+
+                        // But let's first check if we didn't retry this job already
+                        // too often.
+                        let actually_permanent = self.tries >= JOB_RETRIES - 1;
+                        if actually_permanent {
+                            // Okay we tried it already quite often. Next time we would schedule
+                            // this job in three weeks, that doesn't make sense. Therefore
+                            // render this message as failed and stop trying.
+                            info!(context, "Smtp-job #{} failed for the {} time. Mark sending the message as failed.", self.job_id, self.tries);
+                            Status::Finished(Err(format_err!(
+                                "SMTP Transient error failed too often: {}",
```

Comment:
This string is untranslatable and shown to the user. (OTOH, the error itself is going to be in English, too)

Reply:
Yes better errors would be nice, but probably also difficult. I don't know how well parsable/standardized the smtp errors are. Of course there is the error number but the part after it is just text. Any ideas on this?

Reply:
Do you have any ideas how we could make this translatable?

Reply:
Add it to the list in `stock.rs` and use `stock_string_repl_str()`. You can e.g. have a look at how `VideochatInviteMsgBody` was made translatable.